### ISSUE TITLE
upgrade failure in 4.7.9 to 4.7.14

### ISSUE THREAD

**Issue by GaneshKandu** [firefly-iii/firefly-iii#2125]
**Bug description**
I am getting Error while upgrading from **4.7.9** to **4.7.14**

```
FireflyIII
installation and upgrade

Please wait...
An unknown error prevented Firefly III from executing the upgrade commands. Sorry. No currency found with code "eyJpdiI6IksyWlZzUk9kMFRlUFwvdFlYdTRrem5BPT0iLCJ2YWx1ZSI6IkNSWnBOUkpVXC9HcjJJeXJzYzg1RWxRPT0iLCJtYWMiOiI2NTk3MDY4ZmU5YTg4YjY3YmIzNmU3NDg2Y2MxNGIxM2E5MTY3MTZmM2ZlMGQ2Y2I3M2Y1MDM5MGVhNTUzZjA5In0="

Please read the official documentation about this, and upgrade by hand.

```

**Steps to reproduce**
upgrade from 4.7.9 to 4.7.14 using web

**Extra info**
System Environment

.| Version
--- | ---
PHP | 7.2.8
MySQL | 5.6.42
Server API  | Apache 2.0

**Screenshotinfo**

![screenshot-webuzo2 web-2019 02 25-12-20-12](https://user-images.githubusercontent.com/12833130/53319850-c7b43900-38f9-11e9-9999-2414e808a892.png)

**Comment by JC5** [comment]
Mmmm, that looks like a bug. I don't think I can fix it from here. You'll have to wait for the next release (upcoming weekend) so I can build a fix and test it. Sorry about that.

**Comment by JC5** [comment]
I've pushed a fix to the develop branch. Are you capable of trying it out?

**Comment by lastlink** [comment]
This may be related, but I had an issue when updating beyond 4.7.6 b/c I had done a rollback from 4.7.7. since I was still on 7.1 at the time. It was because there wasn't actually a `down() `implementation of the 477 migration.

https://github.com/firefly-iii/firefly-iii/blob/master/database/migrations/2018_09_05_195147_changes_for_v477.php

**Comment by GaneshKandu** [comment]
Hi,

Just Checked with ```firefly-iii-develop``` zip but still getting same issue in upgrade

```

FireflyIII
installation and upgrade

Please wait...
An unknown error prevented Firefly III from executing the upgrade commands. Sorry. No currency found with code "eyJpdiI6IncwNzhJYUVhZ1ZXcWw5R0IxS1wvbGZBPT0iLCJ2YWx1ZSI6IlF6anhNRkJ0djB4a2VyVnlYeTBSaWc9PSIsIm1hYyI6ImIxOTE3ODc0MTNjNzQ1YjUwYWVmZGExOWM4YmEwNDg0YzVlMGZjMTU5YzhkNWRlMTlkMGE5ZGE2Mjk5M2E2NjEifQ=="

Please read the official documentation about this, and upgrade by hand.
```

**Comment by GaneshKandu** [comment]
In old version of Firefly-iii ```data``` column is encrypted but in new version its unencrypted,
Firefly trying to search in ```preferences``` table with encrypted ```data``` where as ```data``` are unencrypted after upgrade

**Comment by JC5** [comment]
Indeed. I hoped my fix would decrypt it in time but clearly it isn't. Let me try again later today.

**Comment by JC5** [comment]
I've pushed another fix. If you could try it, that would be awesome!

**Comment by GaneshKandu** [comment]
Sorry But Still facing same issue

```

FireflyIII
installation and upgrade

Please wait...
An unknown error prevented Firefly III from executing the upgrade commands. Sorry. No currency found with code ""EUR""

Please read the official documentation about this, and upgrade by hand.
```

**Comment by JC5** [comment]
That's OK. I am afraid I have to take baby steps here. I'm happy with your help!

Allow me to try again tonight, OK?

**Comment by JC5** [comment]
I pushed another fix to develop.

**Comment by GaneshKandu** [comment]
Hi @JC5 ,

Tried new fix but still getting same error

> **This below changes worked for me**

added this php code to trim ```$currencyCode``` with ```"``` at line no. 150 of  ```app/Console/Commands/UpgradeDatabase.php```

Line 148-156
```php
            $currencyCode = $this->tryDecrypt($currencyPreference->data);
			
	    $currencyCode = trim($currencyCode,'"'); // trimming  currencyCode with "
			
            $currency = TransactionCurrency::where('code', $currencyCode)->first();
            if (null === $currency) {
                $this->line('Fall back to default currency in migrateBillsToRules().');
                $currency = app('amount')->getDefaultCurrencyByUser($user);
            }
```
Change according to you in [firefly-iii](https://github.com/firefly-iii/firefly-iii/tree/develop) development branch

**Comment by JC5** [comment]
Thanks, I’ll check it out.

**Comment by JC5** [comment]
Pushed another fix. I know what I broke. I'm using `json_decode` since it's consistent with how the value was saved.

**Comment by GaneshKandu** [comment]
Hi @JC5
Thanks for quick fix
i will let you know on monday , today is my holiday

**Comment by GaneshKandu** [comment]
Hi @JC5
Thanks for quick fix
i will let you know on monday , today is my holiday

**Comment by GaneshKandu** [comment]
Hi @JC5
Thanks for quick fix
i will let you know on monday , today is my holiday

**Comment by JC5** [comment]
I was able to test this, and it seems fixed for me.
